### ISSUE TITLE
Cocoon sign-in does not persist

### ISSUE THREAD

**Issue by tvolkert** [flutter/flutter#62346]
The sign-in button seems to work on the build dashboard, and I'm authenticated after signing in... but when I refresh the page, I need to sign-in again.

**Comment by ditman** [comment]
I think there was a change in cocoon where the attempt to "silent sign in" at first was removed, because that may cause an exception in the plugin that would break the onclick of the Sign In button.

(This might trace back to a bug in the Google Sign In web plugin, that I haven't had time to look at yet, the only repro we have for now is to try to sign in with 3rd party cookies disabled either via the browser or an extension)

**Comment by tvolkert** [comment]
This problem runs deeper than just losing the credentials when you refresh the page.  It appears that the authentication credentials expire while your page is loaded (presumably because the OAuth token expires at 5 minutes), meaning you *have* to refresh the page to get validate your credentials.

**Comment by tvolkert** [comment]
FYI @ferhatb

**Comment by ditman** [comment]
I've just posted a PR with some changes to the google sign in plugin that should make this issue more debuggable:

https://github.com/flutter/plugins/pull/2943

The new code should throw more predictable PlatformExceptions in the methods of the plugin, maybe applying the code may give more clues as to why sign in is not working as expected with cocoon.

**Comment by KyleFin** [comment]
I've been unable to reproduce the issue, so maybe this was resolved by other bug fixes. Was this the sign-in button at https://flutter-dashboard.appspot.com/#/build ?

In Chrome I was wondering if my user profile was automatically signing me in again on refresh, but I've also failed to reproduce the issue on Safari. (I signed in, waited >5 min and still seemed to be logged in, refreshed and was still logged in.)

Also, as mentioned in the [dashboard user guide](https://github.com/KyleFin/cocoon/blob/master/app_flutter/USER_GUIDE.md#why-am-i-not-able-to-download-logs), it seems you can access LUCI logs now without signing in. Poking around as someone unfamiliar with the dashboard I wasn't able to find anything that was obviously broken due to sign in issues.

**Comment by ditman** [comment]
@KyleFin thanks for taking a look! I haven't been able to reproduce these issues either. I think there's something about `@google` user accounts that might not trigger the same checks than other users' accounts. It might also be the corporatized version of Chrome.

Next time I look into this, I'll try with a new `@gmail` account in my personal machine, and see if the issue does come up!

**Comment by jmagman** [comment]
Usually what I see these days is I sign in with my `@google` account, close the tab, then awhile later I open the dashboard in a new tab.  I click "Sign In" a few times and nothing happens.  Then I refresh, same thing, "Sign In", nothing happens.  Maybe 2-3 page refreshes later it loads looking like I'm actually signed in, without ever needing to re-enter my credentials.

I recorded it, though I only needed to refresh once to see my avatar appear.  Usually a few refreshes are required (I just saw that last night).

https://user-images.githubusercontent.com/682784/114228506-614cd200-992b-11eb-880b-80ed7f7f86c5.mov

**Comment by KyleFin** [comment]
@jmagman can you confirm there are still use-cases that require being signed in? (Is this just a quirk now or is it actually interfering with workflows?)

**Comment by KyleFin** [comment]
I did observe some strange behavior now on a personal computer with an @gmail account. I clicked the sign-in button and signed in with my account, but now after refreshing about 10 times I still haven't seen my profile picture replace the sign-in button. If I click sign-in again there's an empty white pop-up window that has a blue spinner for a few seconds then disappears.

**Comment by jmagman** [comment]
> @jmagman can you confirm there are still use-cases that require being signed in? (Is this just a quirk now or is it actually interfering with workflows?)

Yes, on that build dashboard it's how tests are re-run to deflake (an unfortunately common occurrence).

**Comment by ditman** [comment]
Maybe this is a race between the flutter app being ready much faster than the google sign in JS SDK? 

> If I click sign-in again there's an empty white pop-up window that has a blue spinner for a few seconds then disappears.

If this happens it means that you're already signed in, and no interaction from you is needed.

I wonder why cocoon (or the web plugin) is not realizing that the user is signed in :/
